src/LodeRunner.Data/Services/BaseService.cs

```diff
@@ -116,33 +115,15 @@ public virtual async Task<HttpStatusCode> Delete(string id)
         /// <param name="entityToCreate">New object.</param>
         /// <param name="cancellationToken">So operation may be cancelled.</param>
         /// <returns>Task<TEntity/> that is the resulting object from the data storage.</returns>
-        public virtual async Task<ApiResponse<TEntity>> Post(TEntity entityToCreate, CancellationToken cancellationToken)
+        public virtual async Task<TEntity> Post(TEntity entityToCreate, CancellationToken cancellationToken)
         {
-            var returnValue = new ApiResponse<TEntity>();
-
-            if (entityToCreate != null && !cancellationToken.IsCancellationRequested && this.CosmosDBRepository.IsCosmosDBReady)
-            {
-                // Update Entity if CosmosDB connection is ready and the object is valid
-                if (this.Validator.ValidateEntity(entityToCreate))
-                {
-                    returnValue.Model = await this.CosmosDBRepository.UpsertDocumentAsync<TEntity>(entityToCreate, cancellationToken);
-                    returnValue.StatusCode = HttpStatusCode.OK;
-                }
-                else
-                {
-                    // TODO: log specific case scenario, even if IsCosmosDBReady() already will do its own logging.
-                    // TODO: log validation errors is any  if not  this.validator.IsValid => this.validator.ErrorMessage
-                    returnValue.Errors = this.Validator.ErrorMessage;
-                    returnValue.StatusCode = HttpStatusCode.BadRequest;
-                }
-            }
-            else
+            // Update Entity if CosmosDB connection is ready
+            if (this.CosmosDBRepository.IsCosmosDBReady)
```

Comment:
is cancellation token handled somewhere else or not needed?

Reply:
It should be handled at a higher level in the controller.